transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -154,4 +155,11 @@ public int pendingTasks() {
     public int registeredChannels() {
         return -1;
     }
+
+    /**
+     * Returns the read-only iterator of {@link Channel}s registered with this {@link EventLoop}.
+     */
+    public Iterator<? extends Channel> registeredChannelsIterator() {
```

Comment:
Most users would not appreciate wildcard generics in return types, lets change to `Iterator<Channel>`